module/zfs/dmu_send.c

```diff
@@ -62,21 +63,26 @@
 int zfs_send_corrupt_data = B_FALSE;
 int zfs_send_queue_length = 16 * 1024 * 1024;
 int zfs_recv_queue_length = 16 * 1024 * 1024;
+/* Set this tunable to FALSE to disable setting of DRR_FLAG_FREERECORDS */
+int zfs_send_set_freerecords_bit = B_TRUE;
```

Comment:
I'll take a closer look at 81f588c specifically, but everything else looked fine on the first pass.